test/expr/closure/anonymous.swift

```diff
@@ -13,38 +13,27 @@ func takesIntArray(_: [Int]) { }
 func takesVariadicInt(_: (Int...) -> ()) { }
 func takesVariadicIntInt(_: (Int, Int...) -> ()) { }
 
-func takesVariadicGeneric<T>(_ f: (T...) -> ()) { }
+func takesVariadicGeneric<T>(_ f: (T...) -> ()) { } // expected-note {{in call to function 'takesVariadicGeneric'}}
 
 func variadic() {
   // These work
 
-  // FIXME: Not anymore: rdar://41416758
   takesVariadicInt({let _ = $0})
-  // expected-error@-1 {{cannot convert value of type '(_) -> ()' to expected argument type '(Int...) -> ()'}}
   takesVariadicInt({let _: [Int] = $0})
-  // expected-error@-1 {{cannot convert value of type '(_) -> ()' to expected argument type '(Int...) -> ()'}}
 
   let _: (Int...) -> () = {let _: [Int] = $0}
-  // expected-error@-1 {{cannot convert value of type '(_) -> ()' to specified type '(Int...) -> ()'}}
 
   takesVariadicInt({takesIntArray($0)})
-  // expected-error@-1 {{cannot pass array of type '[Int]' as variadic arguments of type 'Int'}}
 
   let _: (Int...) -> () = {takesIntArray($0)}
-  // expected-error@-1 {{cannot pass array of type '[Int]' as variadic arguments of type 'Int'}}
 
   takesVariadicGeneric({takesIntArray($0)})
-  // expected-error@-1 {{cannot pass array of type '[Int]' as variadic arguments of type 'Int'}}
-
-  // FIXME(diagnostics): Problems here are related to multi-statement closure bodies not being type-checked together with
-  // enclosing context.
 
+  // FIXME: Problem here is related to multi-statement closure body not being type-checked together with
```

Comment:
Is this really a FIXME to be fixed, or just a consequence of our language rules? I think when I write this test case originally I did not even realize this was a multi-statement closure.

Reply:
I think we could do something in the solver and/or diagnostics to produce a better message here because talking about generic parameter here is not really helpful.